File: src/abc207/abc207_d/main.jl
function match(n::Int, s::Vector{ComplexF64}, t::Vector{ComplexF64})::Bool
    if all(angle.(s) .≈ 0)
        return all(s .≈ t)
    end

    for i in 1:n
        # assume s[1] |-> t[i]
        θ = angle(t[i] / s[1])
        
        res = true
        for j in 1:n
            # s[j] -> ∃t[k]
            sj = (cos(θ) + im * sin(θ)) * s[j]
            res &= any(t .|> tk -> sj .≈ tk)
        end
        
        if res
            return true
        end
    end
    
    return false
end

function sort_by_angle(ps::Vector{ComplexF64})::Vector{ComplexF64}
    g = sum(ps) / length(ps)
    (ps .- g) # |> ps -> sort(ps, by=e -> angle(e))
end

read_int = readline() |> s -> parse(Int, s)
read_ints = readline() |> split .|> e -> parse(Int, e)

function main()
    n = read_int()
    s::Vector{ComplexF64} = []
    t::Vector{ComplexF64} = []
    for _ in 1:n
        a, b = read_ints()
        push!(s, complex(a, b))
    end

    for _ in 1:n
        c, d = read_ints()
        push!(t, complex(c, d))
    end

    s, t = sort_by_angle(s), sort_by_angle(t)
    println(ifelse(match(n, s, t), "Yes", "No"))
end

main()

File: src/abc207/abc207_d/main.py
import numpy as np
# from icecream import ic

def barycenter(poses: list):
    """ poses: List[np.ndarray] の重心 """
    return np.average(np.array(poses), axis=0)


def rot(p: np.ndarray, theta: float, c: np.ndarray = np.zeros((2, 1))):
    """ 点 p: np.ndarray(2x1) を 点 c: np.ndarray(2x1) を中心に theta[rad] 回転する """
    r = np.array([
        [np.cos(theta), -np.sin(theta)],
        [np.sin(theta), np.cos(theta)],
    ])
    return r @ (p - c) + c

def match(n, s: list, t: list):
    """ O(n^3) """
    atol = 1e-6
    for i in range(n):
        # assume s[0] map to t[i] by rotating θ
        theta = (np.arctan2(t[i][1], t[i][0]) - np.arctan2(s[0][1], s[0][0])).item()
        res = True
        for j in range(n):
            # s[j] -> ∃t[k]
            sj = rot(s[j], theta)
            res &= any(map(lambda tk: np.all(np.isclose(sj, tk, atol=atol)), t))
        
        if (res):
            return True
    
    return False

def solve(n, s, t):
    s -= barycenter(s)
    t -= barycenter(t)
    return match(n, s, t)

if __name__ == "__main__":
    # ic.disable()

    n = int(input())
    s, t = [], []
    for i in range(n):
        a, b = list(map(int, input().split()))
        p = np.reshape(np.array([a, b]), (2, 1))
        s.append(p)

    for i in range(n):
        c, d = list(map(int, input().split()))
        p = np.reshape(np.array([c, d]), (2, 1))
        t.append(p)

    print('Yes' if solve(n, s, t) else 'No')